src/data-sources/components/SlugInput.tsx

```diff
@@ -15,12 +15,14 @@ interface SlugInputProps {
 
 export const SlugInput: React.FC< SlugInputProps > = ( { slug, onChange, uuid } ) => {
 	const { slugConflicts, checkSlugConflict } = useDataSources( false );
+	const [ newSlug, setNewSlug ] = useState( slug );
 
 	// eslint-disable-next-line @typescript-eslint/no-misused-promises
 	const debouncedCheckSlugConflict = useDebounce( checkSlugConflict, 500 );
-	const onSlugChange = ( newSlug: string | undefined ): void => {
+	const onSlugUpdate = () => {
```

Comment:
Theoretically we could make this an uncontrolled text input and use the parameter to this function to get `e.target.value` instead of having to update state on every `onChange` event. Really no need for the `newSlug` `useState` at all if we do it that way which reduces rerenders.

```suggestion
	const onSlugUpdate: React.ComponentProps< typeof TextControl >[ 'onBlur' ] = e => {
```

However, for some reason, the `TextControl` component we're using from `@wordpress/components` forces you to have an `onChange` and a `value` prop... so they seem to really want you to use it as a controlled component.

So maybe just ignore all of this. I'd be interested to know why the `value` and `onChange` props are required though.